spec/lib/catalog/submit_order_item_spec.rb

```diff
@@ -101,4 +101,82 @@
       expect { submit_order.process }.to raise_exception(Catalog::InvalidSurvey)
     end
   end
+
+  context "with multiple order items" do
```

Comment:
> It is very complicated to setup a fully loaded portfolio_item that will not break the test.

Right, this was my fear. When it's very difficult to set up the test, to me it indicates that the code is doing too much and should be refactored out into a smaller piece. I think if we did that, it would make testing much easier since we would be able to pass objects along and test for that instead of testing for specifics.

Here's my idea: https://gist.github.com/eclarizio/154d1b8bc8bfa7da62b399e27a68b3ef.

This separates the concerns a little bit, makes the specs a lot easier to digest, and even has the benefit of not needing a change in the `ApprovalTransition` class, since it would then continue to use `SubmitOrder` instead of switching to `SubmitOrderItem`.

Would like to hear thoughts on this

Reply:
The difficulty (rather complexity)  is not for testing the logic of multiple items, but to test submitting of one item. 

To have a new SubmitOrder class would require to call it also in https://github.com/RedHatInsights/catalog-api/pull/805/files#diff-e41c5ceebd417386921147339c918f21R47. It would be strange to submit the order again and again after each item is completed. I was thinking rename `SubmitOrderItem` to `SubmitNextOrderItem` is more meaningful. 

I think I can further simplify the testing for multiple items.

Reply:
@bzwei Sure, in my diff, the `SubmitOrder` class could just be renamed to `SubmitNextOrderItem`, since that is the logic that it is responsible for. My thought with the separation was more so that it's easier to test and follow those specs, which again, to me indicates that the code is cleaner and more easily changed later down the line as well.